diff --git a/src/cx_Logging.c b/src/cx_Logging.c
index 377bd6e..7881ebf 100644
--- a/src/cx_Logging.c
+++ b/src/cx_Logging.c
@@ -2329,7 +2329,11 @@ static PyObject* LogExceptionForPython(
 
     // determine thread state and exception
     threadState = PyThreadState_Get();
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 7
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 11 
+    excType = NULL;
+    excValue = threadState->exc_state.exc_value;
+    traceback = NULL;   
+#elif PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 7
     excType = threadState->exc_state.exc_type;
     excValue = threadState->exc_state.exc_value;
     traceback = threadState->exc_state.exc_traceback;
